### ISSUE TITLE
Tests fail on Windows because of hard-coded line endings

### ISSUE THREAD

**Issue by alysbrooks** [lambdaisland/kaocha#205]
A number of tests fail when running directly on Windows because they have hard-coded newlines ('\n') but Windows uses a different convention ('\r\n').

It looks like `String.format` (and by extension, `format`) has a built-in newline character that expands to the correct one for the format.

[windows_log2.txt](https://github.com/lambdaisland/kaocha/files/6104643/windows_log2.txt)

**Comment by alysbrooks** [comment]
I've come up with a few solutions, and I'm not sure which is best. 

1. Create a `normalize-endings` function and add it to all the tests.
1. Replace `=` with a function that does normalization
1. Instead of comparing strings, compare vectors of strings, each string representing a line.
1. Add normalization to `with-test-out-str`
1. Add normalization to a variant of `with-test-out-str`
1. Use `format`
2. Use `System/lineSeparator`
   
I think 6 is good for tests already using `format` and 7 is good for tests that check against only a separator (which might just be the one). I'm leaning toward using 5 whenever possible and maybe 1 in the remaining cases.

**Comment by plexus** [comment]
How about forcing it to give consistent behavior? e.g. changing `bin/kaocha` to:

```
#!/usr/bin/env bash

clojure -J-Dline.separator=$'\n' -A:dev:test -m kaocha.runner "$@"
```

**Comment by alysbrooks** [comment]
Wow, that's much simpler, thanks! I implemented it in #207

**Comment by alysbrooks** [comment]
Merged!
